hypothesis-python/src/hypothesis/internal/observability.py

```diff
@@ -139,6 +284,10 @@ def make_testcase(
     from hypothesis.core import reproduction_decorator
     from hypothesis.internal.conjecture.data import Status
 
+    # We should only be sending observability reports for datas that have finished
+    # being modified.
+    assert data.frozen
+
     if status_reason is not None:
```

Comment:
`data` needs to be frozen to access` data.spans`, and at that point it seems reasonable to require that all datas passed here are frozen. I don't think freezing more than we used to at any of the callers is harmful, and anyway the whole point of `.freeze` is a phase-transition at which point it cannot be modified, which we definitely expect if we're reporting an observation for a data.